src/PrpShop/PRP/Avatar/QAvLadderMod.cpp

```diff
@@ -0,0 +1,94 @@
+/* This file is part of PlasmaShop.
+ *
+ * PlasmaShop is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PlasmaShop is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PlasmaShop.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "QAvLadderMod.h"
+
+#include <QRadioButton>
+#include <QGridLayout>
+#include <QLabel>
+
+/* QAvLadderMod */
+
+QAvLadderMod::QAvLadderMod(plCreatable* pCre, QWidget* parent)
+    : QCreatable(pCre, kAvLadderMod, parent)
+{
+    plAvLadderMod* mod = plAvLadderMod::Convert(fCreatable);
+
+    fType = new QComboBox(this);
+    fType->addItem("Big", plAvLadderMod::kBig);
+    fType->addItem("FourFeet", plAvLadderMod::kFourFeet);
+    fType->addItem("TwoFeet", plAvLadderMod::kTwoFeet);
+
+    fLoops = new QIntEdit(this);
+    fEnabled = new QCheckBox(tr("Enabled"), this);
+    fGoingUp = new QButtonGroup(this);
+    fLadderView = new QVector3(this);
+
+
+    QGridLayout* layout = new QGridLayout(this);
+    layout->setContentsMargins(8, 8, 8, 8);
+
+    layout->addWidget(fEnabled, 1, 3, 1, 3, Qt::AlignRight);
+    fEnabled->setChecked(mod->isEnabled());
+
+    layout->addWidget(new QLabel(tr("Type:"), this), 1, 0);
+    layout->addWidget(fType, 1, 1, 1, 2);
+    int typeIdx = fType->findData(mod->getType());
+    if (typeIdx < 0) {
+        fType->addItem(QString("Unknown Value: %1").arg(mod->getType()), mod->getType());
+        fType->setCurrentIndex(fType->count() - 1);
+    }
+    else {
```

Comment:
:trollface: I've now fixed this in Visual Studio's Formatting Options, so hopefully it won't go and ruin these behind my back any more. :wink: